user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -205,25 +207,34 @@ public final CountDownLatch reconcileAll(String trigger, String namespace, Label
                 .collect(Collectors.toSet());
         log.debug("reconcileAll({}, {}): Other resources with labels {}: {}", RESOURCE_KIND, trigger, resourceSelector, resourceNames);
 
-        Set<String> usersWithAcls = aclOperations.getUsersWithAcls();
-        log.debug("reconcileAll({}, {}): User with ACLs: {}", RESOURCE_KIND, trigger, usersWithAcls);
+        CountDownLatch outerLatch = new CountDownLatch(1);
 
-        desiredNames.addAll(resourceNames);
-        desiredNames.addAll(usersWithAcls);
+        vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+            future -> {
+                Set<String> usersWithAcls = aclOperations.getUsersWithAcls();
+                future.complete(usersWithAcls);
+            }, res -> {
+                log.debug("reconcileAll({}, {}): User with ACLs: {}", RESOURCE_KIND, trigger, res.result());
+                desiredNames.addAll((Collection<? extends String>) res.result());
+                desiredNames.addAll(resourceNames);
 
-        // We use a latch so that callers (specifically, test callers) know when the reconciliation is complete
-        // Using futures would be more complex for no benefit
-        CountDownLatch latch = new CountDownLatch(desiredNames.size());
+                // We use a latch so that callers (specifically, test callers) know when the reconciliation is complete
+                // Using futures would be more complex for no benefit
+                AtomicInteger counter = new AtomicInteger(desiredNames.size());
 
-        for (String name: desiredNames) {
-            Reconciliation reconciliation = new Reconciliation(trigger, ResourceType.USER, namespace, name);
-            reconcile(reconciliation, result -> {
-                handleResult(reconciliation, result);
-                latch.countDown();
+                for (String name: desiredNames) {
+                    Reconciliation reconciliation = new Reconciliation(trigger, ResourceType.USER, namespace, name);
+                    reconcile(reconciliation, result -> {
+                        handleResult(reconciliation, result);
+                        counter.getAndDecrement();
+                        if (counter.get() == 0) {
```

Comment:
You replace these two calls to decrement and then test with the single call: `if (count.decrementAndGet() == 0) ...`